src/uct/tcp/tcp_ep.c

```diff
@@ -714,16 +739,24 @@ ucs_status_t uct_tcp_ep_connect_to_ep(uct_ep_h tl_ep,
 
     if (!uct_tcp_cm_ep_accept_conn(ep)) {
         ucs_assert(ep->conn_state == UCT_TCP_EP_CONN_STATE_CLOSED);
-        /* EP that are created as CONNECT_TO_EP has to be full-duplex, set RX
+        /* EPs which are created as CONNECT_TO_EP has to be full-duplex, set RX
          * capability as well as TX (that's set in uct_tcp_ep_connect()) */
         uct_tcp_ep_add_ctx_cap(ep, UCT_TCP_EP_FLAG_CTX_TYPE_RX);
 
         uct_tcp_ep_ptr_map_del(ep);
 
-        /* Use remote peer connection sequence number value, since the EP has to
-         * send the CONN_REQ to the peer has to find its EP in the EP PTR map */
+        /* Use the peer's EP PTR map key, since the EP has to send the CONN_REQ
+         * to the peer which has to find an EP created for this connection in
+         * the EP PTR map */
         ep->cm_id.ptr_map_key = addr->ptr_map_key;
         return uct_tcp_ep_connect(ep);
+    } else if (uct_tcp_ep_ptr_map_get(iface, ep->cm_id.ptr_map_key) == NULL) {
```

Comment:
is it possible we just didn't receive a connect request (SYN) from remote side yet, at his point, and this is the reason ptr_map_get() would return NULL?

Reply:
if we didn't receive a connection request from a peer for some reason, it has to find the EP in PTR MAP (the found EP will be equal to an EP for which `uct_ep_connect_to_ep()` was called)
Also, during debugging the issue I saw that we were connected with the peer (so, our EP was deleted from PTR MAP) and disconnected then